packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -211,44 +221,77 @@ export class ResourceBtmService {
     const usedDuplicatedEntities = new Set<string>();
 
     const originalResource = await this.getResourceDataForBtm(resourceId);
-    const originalResourceEntitiesSet = new Set(
-      originalResource.entities.map((entity) => entity.name)
+    const originalResourceEntityNames = originalResource.entities.map(
+      (entity) => entity.name
     );
+    const originalResourceEntityNamesSet = new Set(originalResourceEntityNames);
+
+    const { uniqueToAIResult, uniqueToOriginalResource } =
+      this.findUniqueEntities(
+        recommendedResourceEntities,
+        originalResourceEntityNames
+      );
+
+    // filter out the tables that the AI result has that the original resource doesn't have
+    promptResultObj.microservices = promptResultObj.microservices.map(
+      (microservice) => ({
+        ...microservice,
+        tables: microservice.tables.filter(
+          (tableName) => !uniqueToAIResult.includes(tableName)
+        ),
+      })
+    );
+
+    // from the original resource, leave only the entities that are in uniqueToOriginalResource
+    const originalResourceTables = originalResource.entities.filter((entity) =>
+      uniqueToOriginalResource.includes(entity.name)
+    );
+
+    // remove duplicates and put on the microservice with the least amount of tables
+    promptResultObj.microservices = promptResultObj.microservices
+      .sort((a, b) => a.tables.length - b.tables.length)
+      .map((microservice) => ({
+        name: microservice.name,
+        functionality: microservice.functionality,
+        tables: microservice.tables.filter((tableName) => {
+          const isDuplicatedAlreadyUsed = usedDuplicatedEntities.has(tableName);
+          if (duplicatedEntities.has(tableName)) {
+            usedDuplicatedEntities.add(tableName);
+          }
+          return (
+            originalResourceEntityNamesSet.has(tableName) &&
+            !isDuplicatedAlreadyUsed
+          );
+        }),
+      }))
+      .filter((microservice) => microservice.tables.length > 0);
+
+    // add the original resource to the microservices
+    promptResultObj.microservices.push({
+      name: `${originalResource.name}Original`,
```

Comment:
This is already almost the needed collision handling - just instead of adding "original" to the original one, add "-new" to the new one...

Reply:
@abrl91 after talking to Yuval about his implementation of moving the AI results into the client' ModelOrganizer, it seems that adding the original service into the microservices property is not needed for the client (which already has the original service anyway) and in fact adding it in this way will break Yuval's implementation. So, what I suggest, is to augment the promptResultObj with a **new** property `originalService` which will have the same structure as the other microservices (name, functionality, tables) but will be outside of the `microservices` property. It will allow the clients of the AI result to easily distinguish the original service from the new ones, and decide per usecase what to do with it.